### PR TITLE
PURCHASE-1060/PURCHASE-1059: iOs Entity Header palette component

### PR DESCRIPTION
**Pull request by lilyfromseattle**
Reusable palette component created to be used for partner card and other follow button components.

Screen Shot 2019-06-11 at 11 49 20 AM

__User story__
As a user, I can tap the “follow” button to follow, or “unfollow” button to unfollow the entity.

__Spec__
https://zpl.io/2Egzy5y

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit e12d5a0e640587bcc80aea4e637aaa05bc537ec5

https://deploy-preview-509--artsy-palette.netlify.com

**Comment by ds300** [comment]
I'm a bit confused about the relationship between the iOS and web versions of `EntityHeader`. This one seems designed to have a big follow button on the right while the web version (and the version in the spec) is designed for a small inline link-style button. Also the spacing is a lot different

image

Do we have design approval to change this component so radically for iOS?

I noticed we already have a very similar component in emission for artists `ArtistListItem`, maybe it makes sense to adapt that one for PURCHASE-1059 and keep it in emission rather than introduce this inconsistency into palette?

**Comment by sweir27** [comment]
This was actually intentional given the desire to have a larger touch target on iOS!

Spacing-wise, this is the spec-- it looks like we may want to add those outer margins.
image

**Comment by samchieng** [comment]
@ds300 in general the component should be exactly the same except for the button. Spacing on the left and right just accounts for the normal margins on iOS, but i wasnt sure if this was already included on the page? If so, we dont need it on the component. The 10px on top and bottom are to account for stacking the cards. The button should be the small size (26px height), which I noticed this wasn't on the palette web version, but was included here: https://zpl.io/2j8Knq6. Is there a way I can help QA?

**Comment by ds300** [comment]
Ah OK thanks for the clarification @samchieng and @sweir27 🙌 

@samchieng maybe you could update the palette spec to include the iOS designs?

**Comment by samchieng** [comment]
@ds300 yes, will do that this week!

File: packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            <Avatar
+              size="xs"
+              width={45}
+              height={45}
+              src={imageUrl}
+              initials={initials}
+            />
+          </Flex>
+        )}
+        <Flex ml="10" flexDirection="column" justifyContent="center">
```
[the remark this answers is not in the document]
Comment (lilyfromseattle) [reply packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx]: Yeah I was going off of the artwork card that was already created so I think the button sizes and margins need to be adjusted for both.

File: packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  href,
+  imageUrl,
+  initials,
+  name,
+  meta,
+  FollowButton,
+  ...remainderProps
+}) => {
+  const containerProps = href
+    ? { color: "black100", noUnderline: true, href }
```
[the remark this answers is not in the document]
Comment (lilyfromseattle) [reply packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx]: Nope my bad

File: packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            <Sans mt="-2" size="3t" color="black60">
+              {meta}
+            </Sans>
+          )}
+        </Flex>
+      </Flex>
+
+      <Flex justifyContent="center">
+        {FollowButton && (
+          <Box width={102} height={34}>
```
[the remark this answers is not in the document]
Comment (lilyfromseattle) [reply packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx]: Like have width and height be passed in as props?

File: packages/palette/src/elements/Avatar/Avatar.ios.tsx [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
 }
-
-/**
- * A circular avatar image component.
- */
-const AvatarImage = styled.Image<AvatarProps>`
-  width: ${props => sizeValue(props).diameter};
-  height: ${props => sizeValue(props).diameter};
-  border-radius: ${props => sizeValue(props).diameter};
-`
```
Comment (damassi) [inline packages/palette/src/elements/Avatar/Avatar.ios.tsx]: @lilyfromseattle - Why remove this item?

**Comment by ds300** [comment]
@damassi Lily and I just paired on some fixes to the avatar component. Do you mind taking a quick look? We ended up using a regular react-native image and setting the border radius using extra math because react-native handles oversized border-radius properties different from web.

File: packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (damassi) [reply packages/palette/src/elements/EntityHeader/EntityHeader.ios.tsx]: Also, I know this has been updated, but we should always avoid making things "pixel perfect" if that means falling outside of the boundaries set by our design system [`space`](https://github.com/artsy/palette/blob/master/packages/palette/src/Theme.tsx#L116-L138) values. `width` can be rounded down to `space(10)` (100px); height down to `space(3)` (30px). Rounding is ok as long as the value rounded to lives inside of space.

**Comment by damassi** [comment]
LGTM! 👍 Was just curious why the primitive was no longer being used (not wedded to our primitives implementation however)

**Comment by artsyit** [comment]
:rocket: PR was released in v4.14.15 :rocket:

**Pull request merged**